### PR TITLE
🎉 New destination: RabbitMQ

### PR DESCRIPTION
**Pull request by lgomezm**
## What
- New destination: RabbitMQ

## How
- Sends data to one RabbitMQ routing key (exchange is optional).
- Username/Password supported for connecting to RabbitMQ
- Supports connecting to RabbitMQ virtual host.
- `check` creates a connection to RabbitMQ, then a channel and succeeds if it is open. It fails otherwise.
- Adds stream name, namespace and emmited_at as headers to the RabbitMQ message. This can be used by consumers to filter by them.
- RabbitMQ message data is serialized as JSON.

## Recommended reading order
1. `destination_rabbitmq/destination.py`
2. `unit_tests/unit_test.py`
3. `integration_tests/integration_test.py`

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector added to connector index like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

**Comment by harshithmullapudi** [comment]
Hey @lgomezm thanks for the contribution. We will look into this soon

**Comment by marcosmarxm** [comment]
Hello @lgomezm ! Sorry to not finished your contribution before the date stipulated in the contest. All contributions made before 15-November are eligible to receive the award. We're trying to review your contribution as soon as possible.

File: airbyte-integrations/connectors/destination-rabbitmq/destination_rabbitmq/destination.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    if virtual_host:
+        params.virtual_host = virtual_host
+    return BlockingConnection(params)
+
+
+class DestinationRabbitmq(Destination):
+    def write(
+            self,
+            config: Mapping[str, Any],
+            configured_catalog: ConfiguredAirbyteCatalog,
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/destination-rabbitmq/destination_rabbitmq/destination.py:40]: you're not using the `configured_catalog` take a look in the kvdb destination how to use this.

File: airbyte-integrations/connectors/destination-rabbitmq/destination_rabbitmq/destination.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    virtual_host = config.get('virtual_host')
+    params = pika.ConnectionParameters(
+        host=config['host'],
+        port=config.get('port') or _DEFAULT_PORT
+    )
+    if username and password:
+        params.credentials = PlainCredentials(username=username, password=password)
+    if virtual_host:
+        params.virtual_host = virtual_host
+    return BlockingConnection(params)
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/destination-rabbitmq/destination_rabbitmq/destination.py:33]:
```suggestion
    '''
    host = config.get('host')
    port = config.get('port')
    username = config.get('username')
    password = config.get('password')
    virtual_host = config.get('virtual_host')
    ssl_enabled = config.get('ssl', True)
    amqp_protocol = 'amqp'
    host_url = host

    if ssl_enabled:
        amqp_protocol = 'amqps'

    if port:
        host_url = host + ':' + port

    params = pika.URLParameters(f'{amqp_protocol}://{username}:{password}@{host_url}/{virtual_host}')
    return BlockingConnection(params)
```

why not use the amqp url? this way I was able to connectot to RabitMQ cloud service.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Requested some changes in your code. Also, I'd like to see some UI screenshots showing you're able to connect a service (Facebook, Hubspot or other) select a subset of streams and sync to RabitMQ.

**Comment by marcosmarxm** [comment]
If possible use a free-account in https://customer.cloudamqp.com/login this will help us using the same test base.

**Comment by lgomezm** [comment]
> Requested some changes in your code. Also, I'd like to see some UI screenshots showing you're able to connect a service (Facebook, Hubspot or other) select a subset of streams and sync to RabitMQ.

Hi @marcosmarxm Here are some screenshots showing the new connector working:
Screen Shot 2021-11-26 at 9 59 05 AM
Screen Shot 2021-11-26 at 9 49 29 AM
Screen Shot 2021-11-26 at 9 51 49 AM
Screen Shot 2021-11-26 at 9 48 11 AM

File: airbyte-integrations/connectors/destination-rabbitmq/destination_rabbitmq/destination.py (hunk 2 above)
Comment (lgomezm) [reply airbyte-integrations/connectors/destination-rabbitmq/destination_rabbitmq/destination.py]: Thank you for this suggestion. I'm not familiar with the Python dependency for RabbitMQ, so I didn't know you could connect this way with it. I've updated it in b3f2cee.

File: airbyte-integrations/connectors/destination-rabbitmq/destination_rabbitmq/destination.py (hunk 1 above)
Comment (lgomezm) [reply airbyte-integrations/connectors/destination-rabbitmq/destination_rabbitmq/destination.py:42]: Updated in b3f2cee to use catalog. Thanks for the heads up!

**Comment by lgomezm** [comment]
@marcosmarxm I've addressed your comments. Please take a look again and let me know if you have any other comment. Thank you for reviewing.

**Comment by marcosmarxm** [comment]
@lgomezm can you share the `python -m pytest integration_tests` output too?

**Comment by lgomezm** [comment]
> @lgomezm can you share the `python -m pytest integration_tests` output too?

Hi @marcosmarxm. Sure, here it is:
integration_tests

**Comment by lgomezm** [comment]
Hi @marcosmarxm. Please let me know if there's anything else on my end I should provide.

**Review by marcosmarxm — APPROVED** [review]
thanks this awesome contribution @lgomezm !

**Pull request merged**